### PR TITLE
Added properties to PersistentVolumeClaimVolumeSource

### PR DESCRIPTION
**Pull request by jaredoconnell**
## Changes introduced with this PR

There are a lot of objects with properties missing. This is one that a user encountered that was missing.

Untested.

---
By contributing to this repository, I agree to the [contribution guidelines](https://github.com/arcalot/.github/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: schema.go [hunk 1]
```diff
@@ -2193,7 +2193,39 @@ var Schema = schema.NewTypedScopeSchema[*Config](
...
+				nil,
+				nil,
+				nil,
+				nil,
+			),
+			"readOnly": schema.NewPropertySchema(
+				schema.NewBoolSchema(),
+				schema.NewDisplayValue(
+					schema.PointerTo("readOnly"),
+					schema.PointerTo("readOnly Will force the ReadOnly setting in VolumeMounts."),
```
Comment (dbutenhof) [inline schema.go:2218]: You don't want to make the more direct and obvious claim that this property marks the volume read-only?

**Review by dbutenhof — APPROVED** [review]
Sure. Nit: the description refers to a single property, but you added two. Without confirming with the documentation, these certainly look plausible.

File: schema.go (hunk 1 above)
Comment (jaredoconnell) [reply schema.go:2218]: It's just copy-pasted from the Golang documentation. Do you want us to use a different description?

Comment (dbutenhof) [reply schema.go:2218]: I generally really dislike this sort of empty self-referential documentation. In this case, "readonly" has obvious implications, but "readonly will force [the] readonly setting" is pretty pointlessly reductive. So, "whatever". If you don't feel like tweaking it, consider it approved.

Comment (jaredoconnell) [reply schema.go:2218]: If this was any more complex, I would add additional documentation. But in this case, it's literally read only. That's as self explanatory as we can get. But this documentation clarifies that it isn't the direct setting, and that it only forces a setting elsewhere, so I will keep it as-is.

**Pull request merged**